test/db/cmd/dwarf

```diff
@@ -1772,17 +1775,15 @@ EOF
 EXPECT=<<EOF
             ; CALL XREFS from dbg.$main @ 0x405237, 0x40542d, 0x405ba3, 0x406e29
             ;-- unit p$mozgkigyo palya(smallint,smallint):
-/ void dbg.palya(SmallInt mag, SmallInt szel);
-|           ; var int64_t var_30h @ stack - 0x30
-|           ; var int64_t var_28h @ stack - 0x28
-|           ; var SmallInt i @ stack - 0x1c
-|           ; arg SmallInt szel @ stack - 0x18
+/ void palya(SmallInt mag, SmallInt szel)
 |           ; arg SmallInt mag @ stack - 0x10
-|           0x00401980      55             push  rbp                   ; void palya(SmallInt mag, SmallInt szel);
+|           ; arg SmallInt szel @ stack - 0x18
+|           ; var SmallInt i @ stack - 0x1c
+|           0x00401980      55             push  rbp                   ; void palya(SmallInt mag, SmallInt szel)
 |           0x00401981      4889e5         mov   rbp, rsp
 |           0x00401984      488d6424d0     lea   rsp, [rsp - 0x30]
-|           0x00401989      48895dd8       mov   qword [var_30h], rbx
-|           0x0040198d      4c8965e0       mov   qword [var_28h], r12
+|           0x00401989      48895dd8       mov   qword [rbp - 0x28], rbx
+|           0x0040198d      4c8965e0       mov   qword [rbp - 0x20], r12
```

Comment:
Because in `rz_core_bin_apply_dwarf` I cleared all previous function variables. Because I feel that the information in DWARF should be more accurate?

Reply:
I'm not sure this is correct though. In particular, I think it makes sense to have the variables also when some DWARF info is available. Moreover, Ghidra/IDA show things differently.

![image](https://github.com/rizinorg/rizin/assets/562321/79590ddf-65cd-4e4f-ab93-17de39d573b5)

![image](https://github.com/rizinorg/rizin/assets/562321/ebe414ad-f144-4d13-a554-a362d3f02692)
(Ghidra seems to assume that SI/DI are mag/szel).

Reply:
It should be fine from DWARF. It's just that RBP+offset is now converted to Stack+offset for compatibility.

```
0x000004de:   DW_TAG_subprogram [5] * (0x0000000b)
                DW_AT_name [DW_FORM_string]	("palya")
                DW_AT_prototyped [DW_FORM_flag]	(0x01)
                DW_AT_calling_convention [DW_FORM_data1]	(DW_CC_GNU_borland_fastcall_i386)
                DW_AT_external [DW_FORM_flag]	(0x01)
                DW_AT_low_pc [DW_FORM_addr]	(0x0000000000401980)
                DW_AT_high_pc [DW_FORM_addr]	(0x0000000000401b37)

0x000004f8:     DW_TAG_formal_parameter [6]   (0x000004de)
                  DW_AT_name [DW_FORM_string]	("mag")
                  DW_AT_location [DW_FORM_block1]	(DW_OP_breg6 RBP-8)
                  DW_AT_type [DW_FORM_ref_addr]	(0x00000000000008f6 "SmallInt")

0x00000504:     DW_TAG_formal_parameter [6]   (0x000004de)
                  DW_AT_name [DW_FORM_string]	("szel")
                  DW_AT_location [DW_FORM_block1]	(DW_OP_breg6 RBP-16)
                  DW_AT_type [DW_FORM_ref_addr]	(0x00000000000008f6 "SmallInt")

0x00000511:     DW_TAG_variable [4]   (0x000004de)
                  DW_AT_name [DW_FORM_string]	("i")
                  DW_AT_location [DW_FORM_block1]	(DW_OP_breg6 RBP-20)
                  DW_AT_type [DW_FORM_ref_addr]	(0x00000000000008f6 "SmallInt")

0x0000051b:     NULL
```

Reply:
Ok Ghidra is completely wrong then.

But I still think we should not delete the variables found during analysis just because we also found some info in DWARF. In this case, for example, those vars are used to save/restore a few registers, so I'd want to have a way to track them with a var name.

Moreover, I think in DWARF you could have the source-code level variables/arguments but the compiler may add additional "variables" (like in this case).

Reply:
If you want to try to preserve both the question is how to handle conflicts between analyzed vars and DWARF ones. It's far, far from being trivial.

Reply:
Why not trivial? I think we can say DWARF takes precedence, if available and enabled with the env var. In this case, rbp-0x28 is not on DWARF, so the analysis of rizin should not be removed IMHO.

Reply:
> Why not trivial? I think we can say DWARF takes precedence, if available and enabled with the env var. In this case, rbp-0x28 is not on DWARF, so the analysis of rizin should not be removed IMHO.

Yes, that sounds reasonable

Reply:
I think this is still very important before merging @XVilka @imbillow

Reply:
@ret2libc it was fixed AFAIK